src/components/ManageAccountsDrawer/components/ImportAccounts.tsx

```diff
@@ -139,31 +163,69 @@ export const ImportAccounts = ({ chainId, onClose }: ImportAccountsProps) => {
     if (!wallet) return
     const accountNumber = accounts.length
     const accountResult = await queryClient.fetchQuery(
-      reactQueries.accountManagement.accountIdWithActivity(accountNumber, chainId, wallet),
+      reactQueries.accountManagement.accountIdWithActivityAndMetadata(
+        accountNumber,
+        chainId,
+        wallet,
+        walletDeviceId,
+      ),
     )
     if (!accountResult) return
     setAccounts(previousAccounts => {
-      const { accountId } = accountResult
-      return [...previousAccounts, { accountNumber, accountId }]
+      const { accountId, accountMetadata, hasActivity } = accountResult
+      return [...previousAccounts, { accountId, accountMetadata, hasActivity }]
     })
-  }, [accounts, chainId, queryClient, wallet])
+  }, [accounts, chainId, queryClient, wallet, walletDeviceId])
 
-  const handleToggleAccountId = useCallback((accountId: AccountId) => {
-    console.log('handleToggleAccountId', accountId)
+  const handleAccountIdActiveChange = useCallback((accountId: AccountId, isActive: boolean) => {
+    setAccountIdActiveStateUpdate(previousState => {
+      return { ...previousState, [accountId]: isActive }
+    })
   }, [])
 
+  // TODO: Loading state
+  const handleDone = useCallback(async () => {
+    // for every new account that is active, fetch the account and upsert it into the redux state
+    for (const [accountId, isActive] of Object.entries(accountIdActiveStateUpdate)) {
+      if (!isActive) continue
+      await dispatch(portfolioApi.endpoints.getAccount.initiate({ accountId, upsertOnFetch: true }))
+    }
+
+    const accountMetadataByAccountId = accounts.reduce(
+      (accumulator, { accountId, accountMetadata }) => {
+        return { ...accumulator, [accountId]: accountMetadata }
+      },
+      {},
+    )
+
+    dispatch(
+      portfolio.actions.upsertAccountMetadata({
+        accountMetadataByAccountId,
+        walletId: walletDeviceId,
+      }),
+    )
+
+    const disabledAccountIds = Object.entries(accountIdActiveStateUpdate)
```

Comment:
Or maybe `inactiveAccountIds` might be more aligned.

Reply:
Agree with this one - the update object is really just an object of disabled `AccountId`s

Reply:
95965902383739e7baf88e5ecdf9957c4637823d